inmem_transport.go

```diff
@@ -248,6 +255,7 @@ func (i *inmemPipeline) decodeResponses() {
 				case i.doneCh <- inp.future:
 				case <-i.shutdownCh:
 					return
+				default: // can't write to doneCh, probably because it filled up. meh.
```

Comment:
This is a little gross. And there might be some case still where this transport leaks the decodeResponses() goroutine and prevents a clean shutdown, though I'm not seeing it lately.

Do we want to update the Transport pipeline interface to remove the doneCh entirely?